### PR TITLE
fix (jkube-kit/enricher) : ImageChangeTriggerEnricher shouldn't generate ImageChange triggers for JIB build strategy (#1532)

### LINKED ISSUE #1532: oc:apply with an image built with jib is not working
### Description
The openshift deploymentConfig is hanging not starting the pod if `oc:build` has been used with the `jib` strategy.

Workaround: remove from the deploymentConfig
```
    triggers:
    - type: ConfigChange
    - imageChangeParams:
        automatic: true
        containerNames:
        - quarkus
        from:
          kind: ImageStreamTag
          name: jkube-jib-quarkus:latest
      type: ImageChange
```

### Info
* Eclipse JKube version : 1.8.0
* Kubernetes / Red Hat OpenShift setup and version : Devsandbox

#### how to reproduce :

1. create your own quay.io repo empty. mine is `quay.io/sunix/jkube-jib-quarkus`, configure access for jkube to quay.io (see documentation), oc login to an openshift cluster (devsandbox for me)
2. configure `quickstarts/maven/quarkus/pom.xml` with the properties (generator name to be adapted)
     ```
         <jkube.build.strategy>jib</jkube.build.strategy>
         <jkube.generator.name>quay.io/sunix/jkube-jib-quarkus</jkube.generator.name>
     ```
3. run `mvn clean package oc:build oc:push oc:resource oc:apply`
4. everything looks ok but ...
    Selection_663
    in the DeploymentConfig status:
    ```
    status:
      latestVersion: 0
      observedGeneration: 1
      replicas: 0
      updatedReplicas: 0
      availableReplicas: 0
      unavailableReplicas: 0
      conditions:
        - type: Available
          status: 'False'
          lastUpdateTime: '2022-05-31T15:30:52Z'
          lastTransitionTime: '2022-05-31T15:30:52Z'
          message: Deployment config does not have minimum availability.
    ```

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

Fixes #1532

No ImageChange trigger should added to DeploymentConfig if BuildStrategy
is JIB.

Signed-off-by: Rohan Kumar <rohaan@redhat.com>

## Type of change

 - [X] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] My code follows the style guidelines of this project
 - [ ] I have performed a self-review of my code
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [X] I have added tests that prove my fix is effective or that my feature works
 - [X] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [X] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/EnricherContext.java [hunk 1]
```diff
@@ -135,4 +136,6 @@ default Optional<String> getDependencyVersion(String groupId, String artifactId)
      * @return the Handler Hub instance.
      */
     HandlerHub getHandlerHub();
+
+    JKubeBuildStrategy getBuildStrategy();
```
Comment (manusa) [inline jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/EnricherContext.java:140]: Maybe this field could be declared on the [`Configuration`](https://github.com/eclipse/jkube/blob/682cc6f25ccf3e95e20832d1396fa3890f3a4e26/jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/model/Configuration.java)

WDYT?

Comment (rohanKanojia) [reply jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/EnricherContext.java]: I added it to Configuration, but I couldn't remove it from JKubeEnricherContext as it's getting initialized there. For now I've removed it from EnricherContext interface

https://github.com/eclipse/jkube/blob/682cc6f25ccf3e95e20832d1396fa3890f3a4e26/jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/JKubeEnricherContext.java#L75

File: jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/JKubeEnricherContext.java [hunk 2]
```diff
@@ -70,13 +71,14 @@ public class JKubeEnricherContext implements EnricherContext {
     @Getter(AccessLevel.NONE)
     private ProcessorConfig processorConfig;
     private HandlerHub handlerHub;
+    private JKubeBuildStrategy jKubeBuildStrategy;
```
Comment (manusa) [inline jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/JKubeEnricherContext.java:74]: this should be removed.

**Review by manusa — APPROVED** [review]

**Pull request merged**